src/braket/experimental/autoqasm/program/program.py

```diff
@@ -322,22 +323,21 @@ def register_args(self, args: list[Any]) -> None:
         """
         for arg in args:
             if isinstance(arg, FreeParameter):
-                self.register_parameter(arg)
+                self.register_parameter(arg.name)
             elif isinstance(arg, FreeParameterExpression):
-                # TODO laurecap: Support for expressions
-                raise NotImplementedError(
-                    "Expressions of FreeParameters will be supported shortly!"
-                )
+                for expression_arg in preorder_traversal(arg._expression.args):
```

Comment:
```
if isinstance(arg, FreeParameterExpression):
    for expression_arg in preorder_traversal(arg._expression.expr_free_symbols):
```

Reply:
Then expression_arg can just be cast to a `str`

Reply:
It gives a warning that `.expr_free_symbols` is deprecated, so I have used `.free_symbols` instead. Much cleaner this way, thank you. I do still have to check that it is a `Symbol`, as `.free_symbols` can contain other items (e.g. constants) which are not `Symbol`s.

Reply:
Can you use this to simplify the if, elif statement? `FreeParameter` should have a `._expression` to use, well.